pdsc/localization.py

```diff
@@ -726,6 +743,64 @@ def __init__(self, metadata):
             corners, 1.0, 1.0, 1
         )
 
+class LrocCdrLocalizer(FourCornerLocalizer):
+    """
+    A localizer for the LROC CDR  observations (subclass of
+    :py:class:`FourCornerLocalizer`)
+    """
+
+    DEFAULT_RESOLUTION_M = 1/3*(1e-6)
+    """
+    Sets the default resolution for lroc CDR localization
+    1/3 of that of HiRISE
+    """
+
+    NORMALIZED_PIXEL_SPACE = False
+
+    def __init__(self, metadata):
+        """
+        :param metadata:
+            "lroc_cdr" :py:class:`~pdsc.metadata.PdsMetadata` object
+        """
+        corners = np.array([
+            [metadata.upper_left_latitude, metadata.upper_left_longitude],
+            [metadata.upper_right_latitude, metadata.upper_right_longitude],
+            [metadata.lower_right_latitude, metadata.lower_right_longitude],
+            [metadata.lower_left_latitude, metadata.lower_left_longitude],
+        ])
+        super(LrocCdrLocalizer, self).__init__(
+            corners, metadata.lines, metadata.samples, 1
+        )
+
+class LrocCdrBrowseLocalizer(FourCornerLocalizer):
+    """
+    A localizer for the LROC CDR  observations (subclass of
+    :py:class:`FourCornerLocalizer`)
+    """
+
+    DEFAULT_RESOLUTION_M = 1/3*(1e-6)*2
+    """
+    Sets the default resolution for lroc CDR localization
+    1/3 of that of HiRISE, plus factor of 2 for browse
+    """
+
+    NORMALIZED_PIXEL_SPACE = False
+
+    def __init__(self, metadata):
+        """
+        :param metadata:
+            "lroc_cdr" :py:class:`~pdsc.metadata.PdsMetadata` object
+        """
+        corners = np.array([
+            [metadata.upper_left_latitude, metadata.upper_left_longitude],
+            [metadata.upper_right_latitude, metadata.upper_right_longitude],
+            [metadata.lower_right_latitude, metadata.lower_right_longitude],
+            [metadata.lower_left_latitude, metadata.lower_left_longitude],
+        ])
+        super(LrocCdrBrowseLocalizer, self).__init__(
+            corners, metadata.lines/2.0, metadata.samples/2.0, 1
```

Comment:
For `metadata.lines/2.0, metadata.samples/2.0`, we might want to consider passing in 2.0 as a parameter instead of hard-coding it here. This is the logic specific to our use case. The LROC CDRs we downloaded are pyramid TIFFs. We chose to use the CDRs at 50% reduced resolution, but if others choose to use other resolutions in the pyramid TIFFs, they should be able to change the parameter passed in the function.

Reply:
Updated so user can supply a downsample amount. Added here:(https://github.com/JPLMLIA/pdsc/blob/lroc/pdsc/localization.py#L893)
and:(https://github.com/JPLMLIA/pdsc/blob/lroc/pdsc/localization.py#L797)